Lib/asyncio/runners.py

```diff
@@ -34,7 +34,7 @@ def run(main, *, debug=False):
             "asyncio.run() cannot be called from a running event loop")
 
     if not coroutines.iscoroutine(main):
-        raise ValueError("a coroutine was expected, got {!r}".format(main))
+        raise ValueError("a coroutine object was expected, got {!r}".format(main))
```

Comment:
I'd add another check before this one for `coroutines.iscoroutinefunction(main)` with a more specific error message.